### ISSUE TITLE
Process finished with exit code -1073741819 (0xC0000005)

### ISSUE THREAD

**Issue by MiaomiaoChen2022** [ddangelov/Top2Vec#258]
Hello, thank you for your project.

I met a problem when I try to use this.
 
I have a corpus of about 2354511. But I don't know why it's always interrupted when "Creating lower dimension embedding of documents"

<img width="1398" alt="image" src="https://user-images.githubusercontent.com/100275023/165004343-ec3acc67-171f-4893-a07c-9fce2c70f2b6.png">

**Comment by ddangelov** [comment]
What environment is this in? Perhaps you are running out of ram, with such a large dataset. If that is the case I would recommend trying to create the model with a random subset of the data then use `add_documents` to add the rest.

**Comment by MiaomiaoChen2022** [comment]
> What environment is this in? Perhaps you are running out of ram, with such a large dataset. If that is the case I would recommend trying to create the model with a random subset of the data then use `add_documents` to add the rest.

Thank you for your reply!

My system is Win10 and the RAM is 64G. I suppose you're right. Because when I use a subset of 770K data. I can get the model. Besides, I find a problem that many people may find when using top2vec or bertopic because of a large dataset. It happens in hdbscan and gives the error as "joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed to un-serialize. Please ensure that the arguments of the function are all picklable." .   This can be solved by installing the latest version of hdbscan (0.8.28 version).

And about the "add_documents",  I remember you said that "If adding a large quantity of documents relative to the current model size,  a new model should be trained for best results." So is it  resonable to use "add_documents" to add the rest?

**Comment by ddangelov** [comment]
Yup as long as you take a random sample of your data that is large enough it should be no problem. The only thing you might loose is some granularity of topics but with that much data that is actually desirable.

**Comment by MiaomiaoChen2022** [comment]
> Yup as long as you take a random sample of your data that is large enough it should be no problem. The only thing you might loose is some granularity of topics but with that much data that is actually desirable.

Wow! Thanks for your quick reply! But I don't know how to loose the granularity of topics? Is there any examples?
And I find the model trained using the random subset of 770k data has 9595 topics. Is that too many?

**Comment by ddangelov** [comment]
The smaller the sample the less granularity. You will have to experiment to find what is optimal.
